### ISSUE TITLE
`declaration-colon-newline-after` writes its line break into the text of a comment holding a colon, and under `postcss-scss` leaves a file that does not parse

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#388]
### What minimal example or steps are needed to reproduce the bug?

`declaration-colon-newline-after` takes every colon standing in front of the value for the declaration's own, including one written inside a comment, and writes its line break there. Its twin `declaration-colon-space-after` does not: [#92](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/92) taught the checker those two share to find the first colon outside comments and strings, and the newline rule, which walks the text itself, never learned it.

```css
a { b/*x:y*/:  x; }
```

Two warnings come out, at `1:9` and at `1:13`, and `--fix` writes a break into the text of the comment:

```css
a { b/*x:
y*/
:  x; }
```

The first of the two is the comment's colon, and the break written for it is the harm; the second is the declaration's own, and the break written for that one is right. A second run reports the second warning again, since the first fix moved the text the second was measured in.

Under `postcss-scss` an inline comment holding a colon leaves a file that does not parse at all:

```scss
a { color: //x:y
 red; }
```

comes back from `--fix` as

```scss
a { color:
 //x
:y
 red; }
```

which the next run reads as `Double colon`. The break written into the comment's text closes it early and leaves `:y` standing as code of its own.

### What minimal configuration is needed to reproduce the bug?

```json
{
	"plugins": [
		"@stylistic/stylelint-plugin"
	],
	"rules": {
		"@stylistic/declaration-colon-newline-after": "always"
	}
}
```

### What did you expect to happen?

A colon inside a comment opens no declaration, so nothing is reported for it and nothing written near it. The declaration's own colon is reported and answered as it is today, and one run of `--fix` settles the file.

### What actually happened?

A break is written into the text of the comment, a second warning stands over the same declaration, and under `postcss-scss` the file no longer parses.

### Notes

Found while closing #371, and it stands on `main` at `7e41962` exactly as it stands on that branch. The two rows it leaves in the sweep that branch carries are the only ones that survive it, and they are byte for byte the same on either side.

The answer is the one `declaration-colon-space-after` already carries, and the two rules do not share it only because the newline rule keeps a walk of its own rather than going through `declarationColonSpaceChecker`.
